packages/app/src/components/time-series-chart/logic/use-gapped-line-annotations.ts

```diff
@@ -73,3 +91,10 @@ function formatLabel(
   const end = formatDateFromSeconds(annotation.end, 'axis');
   return `${start} - ${end}: ${annotation.label}`;
 }
+
+function createIsValidCheck(value: unknown) {
+  if (isBoolean(value)) {
+    return (val: unknown) => val === true;
+  }
+  return (val: unknown) => isPresent(val);
+}
```

Comment:
I don't think the check on the first value (`first(values)?.[property]`) makes a lot of sense? The `createIsValidCheck` could be simplified to:

```suggestion
function hasValue(value: unknown) {
  return isBoolean(value) ? val === true : isPresent(val)
}
```

Reply:
Why doesn't it make sense? I only have to check the value once.

Reply:
It complicates code while it probably won't improve performance

Reply:
> it's not going to happen that the first item is a boolean and the second a number

and that ☝️  is an assumption, in theory we could create a schema which is of type `boolean | number`.